code/controllers/subsystem/processing/day_night.dm

```diff
@@ -0,0 +1,131 @@
+
+
+/**
+ * Day Night Cycle Subsystem
+ *
+ * STATION TIMES ARE 24 HR FORMAT
+ * This subsystem simply ticks forward station time by a set increment, and allows handling of time changes and checks.
+ */
+
+SUBSYSTEM_DEF(day_night)
+	name = "Day/Night Cycle"
+	wait = 6 SECONDS // Every 6 seconds, the clock moves forward 1 minutes
+	init_order = INIT_ORDER_DAY_NIGHT
+	/// The current hour
+	var/current_hour = 0
+	/// The current minute
+	var/current_minute = 0
+	/// A list of all currently loaded day night cycle controllers to be handled
+	var/list/cached_controllers = list()
+	/// The amount of time we add every tick
+	var/tick_time = DAY_NIGHT_SUBSYSTEM_FIRE_INCREMENT
+
+/datum/controller/subsystem/day_night/Initialize(start_timeofday)
+	current_hour = rand(0, 23) // We set the starting station time to something random.
+	load_day_night_controller()
+	update_controllers(current_hour)
+	return ..()
```

Comment:
This is going to return SS_NO_INIT. You don't want to call `..()` from subsystem initialize IIRC.